src/redjubjub.rs

```diff
@@ -0,0 +1,133 @@
+//! Implementation of RedJubjub, a specialization of RedDSA to the Jubjub curve.
+//! See section 5.4.6 of the Sapling protocol specification.
+
+use pairing::Field;
+use rand::Rng;
+
+use jubjub::{FixedGenerators, JubjubEngine, JubjubParams, Unknown, edwards::Point};
+use util::hash_to_scalar;
+
+fn h_star<E: JubjubEngine>(a: &[u8], b: &[u8]) -> E::Fs {
+    hash_to_scalar::<E>(b"Zcash_RedJubjubH", a, b)
+}
+
+pub struct Signature<E: JubjubEngine> {
+    r: Point<E, Unknown>,
```

Comment:
It does not (or even large order). The spec explicitly says not. It doesn't in EdDSA either.

The verification equation is [S] G = R + [H*(R || M)] vk, where G is the base and vk is the verifying key. For the binding signature, vk is calculated in a way that ensures it is in the prime order subgroup (it might be the zero point, theoretically, but that would break the Pedersen hash). For the spend authorization signature we don't currently have any checks other than that vk is a valid point.